File: app-sdk/src/main/java/code/sdk/bridge/BridgeCallback.java
package code.sdk.bridge;

import java.io.File;

public interface BridgeCallback {
    void loadUrl(String url);

    void openUrlByBrowser(String url);

    void openUrlByWebView(String url);

    void openApp(String target, String fallbackUrl);

    void goBack();

    void close();

    void refresh();

    void clearCache();

    void saveImage(String url);

    void saveImageDone(boolean succeed);

    void savePromotionMaterialDone(boolean succeed);

    void synthesizePromotionImage(String qrCodeUrl, int size, int x, int y);

    void synthesizePromotionImageDone(boolean succeed);

    void onHttpDnsHttpResponse(String requestId, String response);

    void onHttpDnsWsResponse(String requestId, String response);

    void shareUrl(String url);

    void loginFacebook();

    void logoutFacebook();

    void preloadPromotionImageDone(boolean succeed);

    void shareToWhatsApp(String text, File file);
}

File: app-sdk/src/main/java/code/sdk/bridge/BridgeCallback.kt
package code.sdk.bridge

import java.io.File

interface BridgeCallback {
    fun loadUrl(url: String?)
    fun openUrlByBrowser(url: String?)
    fun openUrlByWebView(url: String?)
    fun openDataByWebView(type: Int, orientation: String?, hover: Boolean, data: String?)
    fun openApp(target: String?, fallbackUrl: String?)
    fun goBack()
    fun close()
    fun refresh()
    fun clearCache()
    fun saveImage(url: String?)
    fun saveImageDone(succeed: Boolean)
    fun savePromotionMaterialDone(succeed: Boolean)
    fun synthesizePromotionImage(qrCodeUrl: String?, size: Int, x: Int, y: Int)
    fun synthesizePromotionImageDone(succeed: Boolean)
    fun onHttpDnsHttpResponse(requestId: String?, response: String?)
    fun onHttpDnsWsResponse(requestId: String?, response: String?)
    fun shareUrl(url: String?)
    fun loginFacebook()
    fun logoutFacebook()
    fun preloadPromotionImageDone(succeed: Boolean)
    fun shareToWhatsApp(text: String?, file: File?)
}
